code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -36,6 +36,65 @@
 	projectile_type = /obj/item/projectile/beam
 	charge_cost = 50
 
+/obj/item/weapon/gun/energy/laser/failure_check(var/mob/living/carbon/human/M)
+	if(damaged)
+		if(projectile_type == "/obj/item/projectile/beam/captain" && prob(5))
+			projectile_type = /obj/item/projectile/beam/lightlaser
+			in_chamber = null
+			in_chamber = new projectile_type(src)
+			fire_delay +=3
+			to_chat(M, "<span class='warning'>Something inside \the [src] pops.</span>")
+			return 1
+		if(projectile_type == "/obj/item/projectile/beam/heavylaser" && prob(15))
+			projectile_type = /obj/item/projectile/beam
+			in_chamber = null
+			in_chamber = new projectile_type(src)
+			fire_sound = 'sound/weapons/Laser.ogg'
+			to_chat(M, "<span class='warning'>Something inside \the [src] pops.</span>")
+			return 1
+		if(projectile_type == /obj/item/projectile/beam && prob(10))
+			projectile_type = /obj/item/projectile/beam/lightlaser
+			in_chamber = null
+			in_chamber = new projectile_type(src)
+			fire_delay +=3
+			to_chat(M, "<span class='warning'>Something inside \the [src] pops.</span>")
+			return 1
+		if(projectile_type == "/obj/item/projectile/beam" && prob(10))
```

Comment:
this is repeated because some guns use /obj/item/projectile/beam and some "/obj/item/projectile/beam" with the quotation marks, and they don't seem interchangable